Src/Common/WebHeaderCollectionExtensions.cs

```diff
@@ -124,6 +126,67 @@ public static IEnumerable<string> GetHeaderValue(this NameValueCollection header
             return null;
         }
 
+        public static void ReadActivityBaggage(this NameValueCollection headers, Activity activity)
```

Comment:
nit: i would add a comment to this method to make it clear how it's parsing headers. adding an example header is helpfull to visualize the internal operations.